xblock/mixins.py

```diff
@@ -516,6 +516,10 @@ def _add_field(self, node, field_name, field):
         as an xml attribute or creates a separate child node.
         """
         value = field.to_string(field.read_from(self))
+
+        if not isinstance(value, basestring):
```

Comment:
The problem with converting `None` to empty strings in general here inside the `_add_field` method is that each field will have to make sure to convert empty strings to `None` values when deserializing, which won't be possible for some field types, for example `String` where an empty string and `None` would be serialize to `""`.